### PR TITLE
SNOW-833537 Separate JWT client

### PR DESCRIPTION
**Pull request by sfc-gh-pfus**
### Description
Please explain the changes you made here.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: dsn.go [hunk 1]
```diff
@@ -169,6 +171,9 @@ func DSN(cfg *Config) (dsn string, err error) {
 	if cfg.ClientTimeout != defaultClientTimeout {
 		params.Add("clientTimeout", strconv.FormatInt(int64(cfg.ClientTimeout/time.Second), 10))
 	}
+	if cfg.JWTClientTimeout != defaultJWTClientTimeout {
+		params.Add("jwtClientTimeout", strconv.FormatInt(int64(cfg.JWTClientTimeout/time.Second), 10))
```
Comment (sfc-gh-dprzybysz) [inline dsn.go:175]: should we validate negative timeout?

Comment (sfc-gh-pfus) [reply dsn.go:175]: I don't like adding in this PR. We have parameters cleanup in our roadmap for the upcoming future.

File: restful.go [hunk 2]
```diff
@@ -92,6 +95,15 @@ func (sr *snowflakeRestful) getFullURL(path string, params *url.Values) *url.URL
 	return ret
 }
 
+func (sr *snowflakeRestful) getClientFor(authType AuthType) *http.Client {
+	switch authType {
```
[the remark this answers is not in the document]
Comment (sfc-gh-pfus) [reply restful.go:100]: Done.

**Review by sfc-gh-pbulawa — APPROVED** [review]

**Review by sfc-gh-ext-simba-hx — APPROVED** [review]
LGTM

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Pull request merged**